src/Features/Core/Portable/CodeFixesAndRefactorings/AbstractFixAllCodeAction.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Roslyn.Utilities;
+using FixAllScope = Microsoft.CodeAnalysis.CodeFixes.FixAllScope;
+
+namespace Microsoft.CodeAnalysis.CodeFixesAndRefactorings
+{
+    /// <summary>
+    /// Fix all code action for a code action registered by
+    /// a <see cref="CodeFixes.CodeFixProvider"/> or a <see cref="CodeRefactorings.CodeRefactoringProvider"/>.
+    /// </summary>
+    internal abstract class AbstractFixAllCodeAction : CodeAction
+    {
+        private bool _showPreviewChangesDialog;
```

Comment:
mutable field is very strange.  but if this is how it already was, then that's ok.